### PR TITLE
fix(argo-cd): Sync ArgoCD helm chart with kustomize

### PR DESCRIPTION
**Pull request by mhoyer**
Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

This is a patch for #645. It syncs some missing parts I mentioned in https://github.com/argoproj/argo-helm/pull/645#issuecomment-815690433 from the original https://github.com/argoproj/argo-cd/blob/v2.0.0/manifests/install.yaml.

### CODE REVIEW

**Comment by mhoyer** [comment]
⚠️ I am not sure about and need some advice:

1. Is this update of the helm chart version in this PR really just a patch? Because of redis image version change. Or better a 3.1.0?
2. We faced some issues on our Tanzu Kubernetes clusters (TKGI) with that redis version 6.2.x not being able to start (not possible to bind to TCP). It seems to be related to the redis docker image and maybe just in constellation with our Kubernetes setup.

**Review by davidkarlsen — APPROVED** [review]

**Comment by davidkarlsen** [comment]
> ⚠️ I am not sure about and need some advice:
> 
> 1. Is this update of the helm chart version in this PR really just a patch? Because of redis image version change. Or better a 3.1.0?

I think it's fine since it's just a third-party and doesn't affect argo as such.

> 2. We faced some issues on our Tanzu Kubernetes clusters (TKGI) with that redis version 6.2.x not being able to start (not possible to bind to TCP). It seems to be related to the redis docker image and maybe just in constellation with our Kubernetes setup.

Is this tied to tanzu / the env, or are additional actions needed?

**Comment by mhoyer** [comment]
> Is this tied to tanzu / the env, or are additional actions needed?

I found the reason, which seems to be an issue with newer Redis 6 and our TKGI installation - but may affect other setups as well. 

It seems with Redis 6 the default TCP binding behavior is [as documented](https://raw.githubusercontent.com/redis/redis/6.0/redis.conf):

> ```
> # By default, if no "bind" configuration directive is specified, Redis listens
> # for connections from all available network interfaces on the host machine.
> ```

As the Redis docker image does not have any default configuration it tries to bind to all interfaces - including the IPv6 interface `::1`. This results in this error we found at the end of our logs:

```bash
$ kubectl logs argocd-redis-59b5c586b6-rsdn2
1:C 08 Apr 2021 14:32:02.255 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 08 Apr 2021 14:32:02.255 # Redis version=6.2.1, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 08 Apr 2021 14:32:02.255 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 08 Apr 2021 14:32:02.256 * monotonic clock: POSIX clock_gettime
1:M 08 Apr 2021 14:32:02.260 # Could not create server TCP listening socket ::*:6379: unable to bind socket, errno: 97
```

I was able to fix it by extending the `args` array inside the template of [`deployment.yaml` of the Redis component](https://github.com/argoproj/argo-helm/blob/3b0ebdda661d8ff73ce2179fcf4adfdcc09c87bb/charts/argo-cd/templates/redis/deployment.yaml#L39) by adding `--bind 127.0.0.1`. Like so:

```yaml
...
      containers:
      - name: {{ template "argo-cd.redis.fullname" . }}
        args:
        - --save
        - "''"
        - --appendonly
        - "no"
        - --bind
        - "127.0.0.1"
```

Now the issue is, that there is currently no good way of fixing this from the "outside" of the chart (not in the template itself). I am actually missing something like the `extraArgs` parameter we already see for the [ArgoCD Server component](https://github.com/argoproj/argo-helm/blob/3b0ebdda661d8ff73ce2179fcf4adfdcc09c87bb/charts/argo-cd/templates/argocd-server/deployment.yaml#L61).

Shall I try to extend this PR with this `extraArgs` capability for `redis` section?

**Comment by ArgTang** [comment]
@mhoyer Could you add a section to the readme to describe where to find the kustomize resources and the command to sync it? Would be helpful for next time :)

**Comment by mhoyer** [comment]
@ArgTang I hope these notes in https://github.com/argoproj/argo-helm/pull/659/commits/7b3d640a6a02a3d3a43d814053e5fab6f2bf291a make sense to you?

**Comment by mhoyer** [comment]
Is there anything I can support one of the maintainers here further?

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: 2.0.0
 description: A Helm chart for ArgoCD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 3.1.1
+version: 3.1.2
```
Comment (oliverbaehler) [inline charts/argo-cd/Chart.yaml:5]:
```suggestion
version: 3.2.2
```

**Review by oliverbaehler — CHANGES_REQUESTED** [review]
Please bump the chart version and we will merge this asap 🍡

**Review by oliverbaehler — APPROVED** [review]
lgtm

**Pull request merged**